compiler/optimizer/LoopCanonicalizer.cpp

```diff
@@ -1964,7 +1964,17 @@ void TR_LoopCanonicalizer::eliminateRedundantInductionVariablesFromLoop(TR_Regio
       //else
       primaryIncr = (int64_t) primaryInductionVariable->getIncrement();
 
+      static bool disableIVEPostDominatorsCheck = (comp()->cg()->hasComplexAddressingMode()
+         || NULL != feGetEnv("TR_DisableIVEPostDominatorsCheck"));
+      TR_PostDominators *postDominators;
+      if (!disableIVEPostDominatorsCheck)
+         {
+         TR_PostDominators computedPostDominators(comp());
+         postDominators = &computedPostDominators;
```

Comment:
Forgive an uninformed query: Why does it make sense to the the address of a local that's about to be destroyed, and the stack space reused for something else?

Reply:
This is a bug waiting to happen. It technically will work today because the JIT compiler never releases allocated memory until the compilation is complete. So the pointer is _technically_ valid after the object goes out of scope because `TR_PostDominators` has an empty destructor. 

However we should *definitely* not rely on such hacky "happens to just work in this case" scenarios. @VermaSh could you fix this? Thanks @keithc-ca for catching that.

Reply:
This is wrong code. Sure the destructor of `TR_PostDominators` is empty, but `postDominators` is going to point to some location on the stack that will no longer be valid once out of the scope of the `if` block.

Reply:
The general paradigm of using containers, such as `TR_PostDominators computedPostDominators(comp());` is that it needs to exist at the outer most scope within which the container is going to be used.

Reply:
Nice catch @keithc-ca

Reply:
I see. I misunderstood `TR_ALLOC`. This needs to get fixed right away then as it can result in undefined behaviour, and given this path is always taken on non-x86 platforms I'd rather not deal with mysterious defects that arise.

Reply:
FWIW, if `TR_PostDominators` has a `TR_ALLOC` specified in its definition, what that means is, you are able to allocate the container doing something like 
```
new (comp->region()) TR_PostDominators(..)
```
or
```
new (comp->trHeapMemory()) TR_PostDominators(..)
```
ie, `TR_ALLOC` expands to a bunch of `placement new` definitions. If the code above was:
```
         postDominators  = new (comp->region()) TR_PostDominators(comp());
```
then there would be no issues. However, the issue is because the container itself was allocated as a local.